plugins/catalog-backend/src/service/DefaultLocationService.ts

```diff
@@ -54,6 +54,34 @@ export class DefaultLocationService implements LocationService {
     return this.store.deleteLocation(id);
   }
 
+  private async getUnprocessedEntities(
+    unprocessed: DeferredEntity[],
+    entities: Entity[],
+  ): Promise<Entity[]> {
+    const currentEntity = unprocessed.pop();
+    if (!currentEntity) {
+      return entities;
+    }
+    const processed = await this.orchestrator.process({
+      entity: currentEntity.entity,
+      state: {}, // we process without the existing cache
+    });
+
+    if (processed.ok) {
+      const { metadata, kind } = processed.completedEntity;
+      if (
+        entities.some(e => e.metadata.name === metadata.name && e.kind === kind)
+      ) {
+        throw new Error(`Duplicate nested entity: ${metadata.name}`);
+      }
+      return await this.getUnprocessedEntities(
```

Comment:
I find that the existing queue based implementation is easier for most people to read tbh, would it be alright to keep that in place rather than a recursive one?

Reply:
Yeah no problem